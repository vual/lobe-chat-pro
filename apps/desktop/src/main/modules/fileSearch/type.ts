import { FileResult, SearchOptions } from '@/types/fileSearch';

/**
 * File Search Service Implementation Abstract Class
 * Defines the interface that different platform file search implementations need to implement
 */
export abstract class FileSearchImpl {
  /**
   * Perform file search
   * @param options Search options
   * @returns Promise of search result list
   */
  abstract search(options: SearchOptions): Promise<FileResult[]>;

  /**
   * Check search service status
   * @returns Promise indicating if service is available
   */
  abstract checkSearchServiceStatus(): Promise<boolean>;

  /**
   * Update search index
   * @param path Optional specified path
   * @returns Promise indicating operation success
   */
  abstract updateSearchIndex(path?: string): Promise<boolean>;
}
