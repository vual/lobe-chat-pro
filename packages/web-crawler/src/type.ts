export interface CrawlSuccessResult {
  content?: string;
  contentType: 'text' | 'json';
  description?: string;
  length?: number;
  siteName?: string;
  title?: string;
  url: string;
}

export interface CrawlErrorResult {
  content: string;
  errorMessage: string;
  errorType: string;
  url: string;
}

export interface FilterOptions {
  // 是否启用Readability
  enableReadability?: boolean;

  pureText?: boolean;
}

type CrawlImplType = 'naive' | 'jina' | 'browserless' | 'search1api';

type CrawlImplParams<T> = T & {
  filterOptions: FilterOptions;
};

export type CrawlImpl<Params = object> = (
  url: string,
  params: CrawlImplParams<Params>,
) => Promise<CrawlSuccessResult | undefined>;

export interface CrawlUrlRule {
  // 内容过滤配置（可选）
  filterOptions?: FilterOptions;
  impls?: CrawlImplType[];
  // URL匹配模式，仅支持正则表达式
  urlPattern: string;
  // URL转换模板（可选），如果提供则进行URL转换
  urlTransform?: string;
}
