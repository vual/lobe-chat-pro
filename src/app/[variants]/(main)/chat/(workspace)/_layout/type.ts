import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  conversation: ReactNode;
  portal: ReactNode;
  topic: ReactNode;
}
