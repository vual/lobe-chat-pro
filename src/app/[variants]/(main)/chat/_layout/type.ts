import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  session: ReactNode;
}
