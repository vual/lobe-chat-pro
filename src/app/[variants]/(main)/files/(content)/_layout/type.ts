import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  modal: ReactNode;
}
