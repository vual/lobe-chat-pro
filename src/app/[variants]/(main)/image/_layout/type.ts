import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  topic: ReactNode;
}
