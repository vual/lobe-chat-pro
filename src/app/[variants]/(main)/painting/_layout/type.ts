import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  palette: ReactNode;
}
