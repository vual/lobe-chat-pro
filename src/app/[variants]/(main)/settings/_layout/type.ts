import { ReactNode } from 'react';

export interface LayoutProps {
  category: ReactNode;
  children: ReactNode;
}
