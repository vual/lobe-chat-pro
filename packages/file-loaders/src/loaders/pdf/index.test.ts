// @vitest-environment node
import path from 'node:path';
import { beforeEach, describe, expect, it } from 'vitest';

import type { FileLoaderInterface } from '../../types';
import { PdfLoader } from './index';

// 确保你已经在 fixtures 目录下放置了 test.pdf 文件
const fixturePath = (filename: string) => path.join(__dirname, `./fixtures/${filename}`);

let loader: FileLoaderInterface;

const testFile = fixturePath('test.pdf');
const nonExistentFile = fixturePath('nonexistent.pdf');

beforeEach(() => {
  loader = new PdfLoader();
});

describe('PdfLoader', () => {
  it('should load pages correctly from a PDF file', async () => {
    const pages = await loader.loadPages(testFile);

    expect(pages.length).toBeGreaterThan(0);

    expect(pages).toMatchSnapshot();
  });

  it('should aggregate content correctly', async () => {
    const pages = await loader.loadPages(testFile);
    const content = await loader.aggregateContent(pages);
    // 默认聚合是以换行符连接各页内容
    expect(content).toMatchSnapshot();
  });

  it('should handle file read errors in loadPages', async () => {
    const pages = await loader.loadPages(nonExistentFile);
    expect(pages).toHaveLength(1); // 即使失败也返回一个包含错误信息的页面
    expect(pages[0].pageContent).toBe('');
    expect(pages[0].metadata.error).toContain('Failed to load or parse PDF file:');
  });

  it('should attach document metadata correctly', async () => {
    // 首先加载页面以初始化 pdfInstance，尽管此方法不直接使用页面
    await loader.loadPages(testFile);
    const metadata = await loader.attachDocumentMetadata!(testFile);

    expect(metadata).toMatchSnapshot();
  });
});
