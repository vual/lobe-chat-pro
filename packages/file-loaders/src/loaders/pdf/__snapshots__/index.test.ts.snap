// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PdfLoader > should aggregate content correctly 1`] = `
"<page pageNumber="1">
简单报告
副标题
轻点或点按此占位符⽂本并开始键⼊即可开始。你可以在 Mac、iPad、iPhone 或
iCloud.com 上查看和编辑此⽂稿。
轻松编辑⽂本、更改字体以及添加精美的图形。使⽤段落样式来使整篇⽂稿保持⼀
致的⻛格。例如，此段落使⽤“正⽂”样式。你可以在“格式”控制的“⽂本”标签⻚中
更改样式。
若要添加照⽚、图像画廊、⾳频⽚段、视频、图表或任意 700 多种可⾃定义形状，
请在⼯具栏中轻点或点按其中⼀个插⼊按钮，或者将对象拖放到⻚⾯中。你可以分
层放置对象、调整其⼤⼩以及将其放在⻚⾯中的任意位置。若要更改对象随⽂本移
动的⽅式，请选择对象并随后轻点或点按“格式”控制中的“排列”标签⻚。
⼩标题
Pages ⽂稿可⽤于⽂字处理和⻚⾯布局。此“简单报告”模板为⽂字处理⽽设置，如
此⼀来，⽂本便会随着你的键⼊⽽从某⼀⻚流向下⼀⻚，到达⻚⾯末尾时会⾃动创
建新的⻚⾯。
在⻚⾯布局⽂稿中，你可以⼿动重新排列⻚⾯并随意调整⻚⾯中的⽂本框、图像和
其他对象的位置。若要创建⻚⾯布局⽂稿，请在模板选取器中选取⼀种⻚⾯布局模
板。你也可以在 Mac、iPad 或 iPhone 上将此⽂稿改为⻚⾯布局，⽅法是在“⽂稿”
控制中关闭“⽂稿正⽂”。
“这是⼀个引⽤（报告中的关键短语）的例⼦。轻点或点按此
⽂本添加你⾃⼰的内容。”
⻚脚 
1
</page>

<page pageNumber="2">
这是第⼆⻚的内容
⻚脚 
2
</page>"
`;

exports[`PdfLoader > should attach document metadata correctly 1`] = `
{
  "pdfInfo": {
    "CreationDate": "D:20250419155028Z00'00'",
    "Creator": "Pages文稿",
    "EncryptFilterName": null,
    "IsAcroFormPresent": false,
    "IsCollectionPresent": false,
    "IsLinearized": false,
    "IsSignaturesPresent": false,
    "IsXFAPresent": false,
    "Language": null,
    "ModDate": "D:20250419155028Z00'00'",
    "PDFFormatVersion": "1.3",
    "Producer": "macOS 版本15.3.2（版号24D81） Quartz PDFContext",
    "Title": "test",
  },
  "pdfMetadata": null,
  "pdfVersion": "4.10.38",
}
`;

exports[`PdfLoader > should load pages correctly from a PDF file 1`] = `
[
  {
    "charCount": 576,
    "lineCount": 23,
    "metadata": {
      "pageNumber": 1,
    },
    "pageContent": "简单报告
副标题
轻点或点按此占位符⽂本并开始键⼊即可开始。你可以在 Mac、iPad、iPhone 或
iCloud.com 上查看和编辑此⽂稿。
轻松编辑⽂本、更改字体以及添加精美的图形。使⽤段落样式来使整篇⽂稿保持⼀
致的⻛格。例如，此段落使⽤“正⽂”样式。你可以在“格式”控制的“⽂本”标签⻚中
更改样式。
若要添加照⽚、图像画廊、⾳频⽚段、视频、图表或任意 700 多种可⾃定义形状，
请在⼯具栏中轻点或点按其中⼀个插⼊按钮，或者将对象拖放到⻚⾯中。你可以分
层放置对象、调整其⼤⼩以及将其放在⻚⾯中的任意位置。若要更改对象随⽂本移
动的⽅式，请选择对象并随后轻点或点按“格式”控制中的“排列”标签⻚。
⼩标题
Pages ⽂稿可⽤于⽂字处理和⻚⾯布局。此“简单报告”模板为⽂字处理⽽设置，如
此⼀来，⽂本便会随着你的键⼊⽽从某⼀⻚流向下⼀⻚，到达⻚⾯末尾时会⾃动创
建新的⻚⾯。
在⻚⾯布局⽂稿中，你可以⼿动重新排列⻚⾯并随意调整⻚⾯中的⽂本框、图像和
其他对象的位置。若要创建⻚⾯布局⽂稿，请在模板选取器中选取⼀种⻚⾯布局模
板。你也可以在 Mac、iPad 或 iPhone 上将此⽂稿改为⻚⾯布局，⽅法是在“⽂稿”
控制中关闭“⽂稿正⽂”。
“这是⼀个引⽤（报告中的关键短语）的例⼦。轻点或点按此
⽂本添加你⾃⼰的内容。”
⻚脚 
1",
  },
  {
    "charCount": 14,
    "lineCount": 3,
    "metadata": {
      "pageNumber": 2,
    },
    "pageContent": "这是第⼆⻚的内容
⻚脚 
2",
  },
]
`;
